lib/cb2/breaker.rb

```diff
@@ -16,7 +17,7 @@ def run
       ret = yield
       process_success
     rescue => e
-      process_error
+      if (!(@whitelist.any? { |klass| e.is_a?(klass) })) then process_error end
```

Comment:
move to separate method called, e.g. error_opens_circuit?(error), in order to document what is happening a little better.